kratos/utilities/read_materials_utility.cpp

```diff
@@ -259,18 +246,144 @@ void ReadMaterialsUtility::AssignPropertyBlock(Parameters Data)
                 table.insert(table_param["data"][i][0].GetDouble(),
                             table_param["data"][i][1].GetDouble());
             }
-            p_prop->SetTable(r_input_var, r_output_var, table);
+            pNewProperty->SetTable(r_input_var, r_output_var, table);
         }
     } else {
-        KRATOS_INFO("Read materials") << "No tables defined for material ID: " << property_id << std::endl;
+        KRATOS_INFO("Read materials") << "No tables defined for material ID: " << pNewProperty->Id() << std::endl;
     }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ReadMaterialsUtility::CreateSubProperties(
+    ModelPart& rModelPart,
+    Parameters Data,
+    Properties::Pointer& pNewProperty
+    )
+{
+    KRATOS_TRY;
+
+    if (Data.Has("sub_properties")) {
+
+        PointerVectorSet<Properties, IndexedObject> list_sub_properties;
+
+        const std::size_t number_of_subproperties = Data["sub_properties"].size();
+
+        // We do a check of ids
+        for(std::size_t i_sub_prop=0; i_sub_prop < number_of_subproperties; ++i_sub_prop) {
+            const int sub_property_id = Data["sub_properties"][i_sub_prop]["properties_id"].GetInt();
+            KRATOS_ERROR_IF(sub_property_id == static_cast<int>(pNewProperty->Id())) << "You cannot assign to a property a subproperty with the same Id" << std::endl;
```

Comment:
I think this is allowed

Reply:
I haven't understand that, OK

Reply:
Lets wait for @KratosMultiphysics/technical-committee to confirm before you change it